src/VisualStudio/Core/Def/Implementation/Library/ObjectBrowser/AbstractObjectBrowserLibraryManager.cs

```diff
@@ -76,7 +76,9 @@ public void Dispose()
                     var oldDocument = e.OldSolution.GetDocument(e.DocumentId);
                     var newDocument = e.NewSolution.GetDocument(e.DocumentId);
 
-                    await DocumentChangedAsync(oldDocument, newDocument).ConfigureAwait(false);
+                    // make sure we do this in background thread. we don't care about ordering of events
+                    // we just need to refresh OB at some point if it ever needs to be updated
+                    Task.Run(() => DocumentChangedAsync(oldDocument, newDocument));
```

Comment:
Ah, i could see htat being the case.  You double-click to open something.  And then there's a hang while OB then just refreshes itself.